src/tiled/terrainbrush.cpp

```diff
@@ -376,6 +377,25 @@ void TerrainBrush::updateBrush(QPoint cursorPos, const QVector<QPoint> *list)
         if (checked[i])
             continue;
 
+        // to support isometric staggered, make edges into variables
+        QPoint upPoint(x, y-1);
+        QPoint bottomPoint(x, y+1);
+        QPoint leftPoint(x-1, y);
+        QPoint rightPoint(x+1, y);
+
+        StaggeredRenderer* renderer = dynamic_cast<StaggeredRenderer*>(mapDocument()->renderer());
+        if (renderer) {
```

Comment:
One more thing here, since you're not going to use `renderer` outside of the `if` body, you should put it in its scope:
```c++
if (auto renderer = dynamic_cast<StaggeredRenderer*>(mapDocument()->renderer())) {
```